tests/unit.rs

```diff
@@ -255,6 +255,30 @@ fn test_form_serialize() {
 }
 
 #[test]
+fn host_and_port_display() {
+    let data = [
+        (HostAndPort{ host: Host::Domain("www.mozilla.org"), port: 80}, "www.mozilla.org:80"),
+        (
+            HostAndPort{ host: Host::Ipv4(Ipv4Addr::new(1, 35, 33, 49)), port: 65535 },
+            "1.35.33.49:65535"
+        ),
+        (
+            HostAndPort{
+                host: Host::Ipv6(Ipv6Addr::new(
+                    0x2001, 0x0db8, 0x85a3, 0x08d3, 0x1319, 0x8a2e, 0x0370, 0x7344
+                )),
+                port: 1337
+            },
+            "[2001:db8:85a3:8d3:1319:8a2e:370:7344]:1337"
+        )
+    ];
+
+    for &(ref input, result) in &data {
```

Comment:
Avoid `for` loops like this for unit tests. When a test fail it will show the line number for `assert_eq!`, but that doesn’t show which of the entries failed.

In this case I think duplicating the `assert_eq!` + `format!` line is fine, since it’s not long and there’s only two test cases. In a case when de-duplicating testing code is desirable, consider using a macro like `assert_from_file_path!` in this file. This makes panic messages use the line number of the macro usage.